### PR TITLE
Report uptime status based on component events

### PR DESCRIPTION
**Pull request by ezequielc**
Report uptime status based on component events.
Also, cleaned up integration test files inherited from financial component and never implemented.

@alex-deaconu Please review. Thanks!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ demo/build
 demo/node_modules
 
 npm-debug.log
+package-lock.json
```
Comment (alex-deaconu) [inline .gitignore]: @ezequielc I've never been able to generate a `package-lock.json` file locally for this repo. Wondering however if we should include this file for reference? Thanks!

File: src/rise-data-weather.js [hunk 2]
```diff
@@ -225,6 +225,20 @@ class RiseDataWeather extends CacheMixin( RiseElement ) {
...
+    case RiseDataWeather.EVENT_REQUEST_ERROR:
+    case RiseDataWeather.EVENT_DATA_ERROR:
+      super._setUptimeError( true );
+      break;
+    case RiseDataWeather.EVENT_DATA_UPDATE:
+      super._setUptimeError( false );
+      break;
+    default:
+    }
+    super._sendEvent( name, detail );
```
Comment (alex-deaconu) [inline src/rise-data-weather.js]: @ezequielc I'm wondering if we should call the `super` function before executing the uptime reporting code, to be consistent with other overrides in the repo. Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc looks good, 2 suggestions inline. Thanks!

**Comment by ezequielc** [comment]
@alex-deaconu Good suggestions, will work on them. Thanks

**Comment by alex-deaconu** [comment]
@ezequielc looks good, thanks for addressing!

**Pull request merged**